### PR TITLE
`azurerm_dev_test_lab`: Deprecate `storage_type`

### PR DESCRIPTION
**Pull request by jiaweitao001**
* Current API does not support `storage_type`, this is causing lots of test failing in team city. 
* No Swagger available for new API version, so deprecating this attribute for good.

### CODE REVIEW

File: internal/services/devtestlabs/dev_test_lab_resource.go [hunk 1]
```diff
@@ -100,6 +101,21 @@ func resourceDevTestLab() *pluginsdk.Resource {
 			},
 		},
 	}
+
+	if !features.FourPointOhBeta() {
+		resource.Schema["storage_type"] = &pluginsdk.Schema{
+			Type:     pluginsdk.TypeBool,
```
Comment (katbyte) [inline internal/services/devtestlabs/dev_test_lab_resource.go:107]: This shouldn't be a bool?

**Review by katbyte — CHANGES_REQUESTED** [review]

File: internal/services/devtestlabs/dev_test_lab_resource.go (hunk 1 above)
Comment (jiaweitao001) [reply internal/services/devtestlabs/dev_test_lab_resource.go]: Right, type fixed.

File: internal/services/devtestlabs/dev_test_lab_resource.go [hunk 2]
```diff
@@ -100,6 +101,21 @@ func resourceDevTestLab() *pluginsdk.Resource {
 			},
 		},
 	}
+
+	if !features.FourPointOhBeta() {
```
Comment (katbyte) [inline internal/services/devtestlabs/dev_test_lab_resource.go:105]: we'll need to wrap the code in the create/read/update funtions to handle this property not existing anymore?

**Review by katbyte — CHANGES_REQUESTED** [review]

File: internal/services/devtestlabs/dev_test_lab_resource.go (hunk 2 above)
Comment (jiaweitao001) [reply internal/services/devtestlabs/dev_test_lab_resource.go:105]: Updated. Thanks.

**Review by katbyte — APPROVED** [review]
Thanks @jiaweitao001 - LGTM 🌻

**Pull request merged**